app/resources/linked_profile_resource.rb

```diff
@@ -0,0 +1,20 @@
+class LinkedProfileResource < BaseResource
+  attribute :token
+  attributes :external_user_id, :url, :share_to, :share_from
+
+  has_one :user
+  has_one :linked_site
+
+  filter :self, apply: -> (_r, _v, options) {
+    current_user = options[:context][:user]
+    records.where(id: current_user&.id) || User.none
+  }
+
+  def fetchable_fields
+    if current_user == _model
+      super
+    else
+      super - :token
```

Comment:
You missed an important detail here: `:token` needs to be `[:token]` — you can't subtract a symbol from an array, but you can subtract an array from an array